tests/hypermodel/test_preprocessor.py

```diff
@@ -150,9 +167,26 @@ def map_func(x):
 def test_feature_engineering_fix_keyerror():
     data = structured_data(100)
     dataset = tf.data.Dataset.from_tensor_slices(data)
-    feature = preprocessor.FeatureEngineering([
-        'categorical', 'categorical', 'numerical', 'numerical', 'categorical',
-        'categorical', 'categorical', 'categorical'])
+    feature = preprocessor.FeatureEngineering(
+        ak.StructuredDataInput(
+            column_names=[
```

Comment:
avoid the duplication by extracting two global variables column_names_for_tests, column_types_for_tests.